posts/2024-05-01-spring-boot-3.adoc

```diff
@@ -0,0 +1,238 @@
+seo-description: In version 23.0.0.9 and later, you can easily run a Spring Boot 3.x application from a WAR file on Open Liberty. This post use the Spring Petclinic sample application and the Liberty Maven plugin to create and run a WAR-based Spring Boot 3.1 application on Open Liberty.
+blog_description: In version 23.0.0.9 and later, you can easily run a Spring Boot 3.x application from a WAR file on Open Liberty. This post uses the Spring Petclinic sample application and the Liberty Maven plugin to create and run a WAR-based Spring Boot 3.1 application on Open Liberty.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Running a Spring Boot 3.x application WAR file on Liberty
+Scott Kurz <https://github.com/scottkurz>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+As of version 23.0.0.9, Liberty offers support for Spring Boot 3.1, allowing developers to take advantage of the Spring and Spring Boot libraries while providing a range of https://openliberty.io/blog/2022/10/17/memory-footprint-throughput-update.html[performance] enhancements specifically engineered to provide an industry-leading, first-class platform for Spring Boot applications.
+
+When packaging a Spring Boot app in the WAR format, developers and administrators get to choose from two Liberty deployment options, each with a different set of advantages.
+
+A Spring Boot WAR can simply be deployed like any standard Jakarta EE WAR using the same developer tooling and server configuration, and providing a common DevOps workflow as with your other Liberty WAR deployments.
+
+In addition, with Liberty, you can use a special https://openliberty.io/docs/latest/deploy-spring-boot.html[optimized deployment for Spring Boot applications]. The optimization adds some configuration options and provides advantages for efficiently building Docker container images by using some extra `pom.xml` and `server.xml` file configuration.
+
+In this blog post, we'll first create a simple app by using the link:https://start.spring.io/[Spring Initializr], and then deploy it as a WAR to Liberty without any modifications, using the same deployment as for any WAR.
+
+We then show how to optimize the Spring Boot deployment by adding the Liberty Maven Plugin (`liberty-maven-plugin`) and a simple `server.xml` configuration with the `springBoot-3.0` feature.   We can then build an efficiently-layered Docker image.
+
+
+== Prerequisites
+- The sample requires Java 21 (though it could be easily reworked using Java 17)
+
+== Deploy as standard WAR
+
+1. Create a project from your browser using link:https://start.spring.io/#!type=maven-project&language=java&platformVersion=3.2.5&packaging=war&jvmVersion=21&groupId=demo&artifactId=spring-liberty&name=spring-liberty&description=Demo%20project%20for%20Spring%20Boot&packageName=demo&dependencies=web[this Spring Initializr configuration].
```

Comment:
I'm not going to suggest the edit from PR review mode, but I think this 1st step should be part of the prereqs- it's the same step for either procedure so better to add as a prereq than ask "optimized deployers" to go back to the option they already skipped over

Reply:
OK,  see why you are saying this, which means the flow isn't as clear as I'd thought.   You are assuming that because these are two parallel options that someone following the blog instructions could do either Option 1 OR Option 2, and therefore could start right at Option 2.

If I did that though I'd have to move the generation of the "2. Add the following Spring web controller file" into the prereq... the Option 2. builds on the Option 1. instructions.

I think it works well enough walking you through both options... it's not like either one is too long in and of itself.

Do you think it's worth trying to make this all more clear somehow, (the article organization I'm explaining in this comment)

I wonder if I should go back and rename the headings since I've since changed them to Option 1... Option 2.